src/terminal.ts

```diff
@@ -0,0 +1,354 @@
+    this.onDidWriteEmitter.fire(clear);
+    this.onDidWriteEmitter.fire(output);
+  }
+}
+
+function stringifyTerminalState(state: DecodeTerminalState): string {
+  const lines = [decodeTerminalTitle];
+  const { params, userInput, decoderResult } = state;
+  let { statusMessage } = state;
+  if (params instanceof Error && !(params instanceof DecodeParamsError)) {
+    lines.push(red(toTerminalEOL(params.message)));
+  } else {
+    const { fqbn, sketchPath } = params;
+    lines.push(
+      `Sketch: ${green(path.basename(sketchPath))} FQBN: ${green(fqbn)}`
+    );
+    if (params instanceof DecodeParamsError) {
+      // error overrules any status message
+      statusMessage = red(toTerminalEOL(params.message));
+    } else {
+      if (userInput) {
+        lines.push('', userInput);
+      }
+      if (decoderResult) {
+        lines.push(
+          '',
+          decoderResult instanceof Error
+            ? red(toTerminalEOL(decoderResult.message))
+            : stringifyDecodeResult(decoderResult)
+        );
+      }
+    }
+    if (statusMessage) {
+      lines.push('', statusMessage, '');
+    }
+  }
+  return stringifyLines(lines);
+}
+
+function stringifyDecodeResult(decodeResult: DecodeResult): string {
+  const lines: string[] = [];
+  if (decodeResult.exception) {
+    const [message, code] = decodeResult.exception;
+    lines.push(red(`Exception ${code}: ${message}`));
+  }
+  const registerLines = Object.entries(decodeResult.registerLocations);
+  for (const [name, location] of registerLines) {
+    lines.push(`${red(name)}: ${stringifyLocation(location)}`);
+  }
+  if (registerLines.length) {
+    lines.push('');
+  }
+  if (decodeResult.stacktraceLines.length) {
+    lines.push('Decoding stack results');
+  }
+  lines.push(...decodeResult.stacktraceLines.map(stringifyGDBLine));
+  if (decodeResult.allocLocation) {
+    const [location, size] = decodeResult.allocLocation;
+    lines.push(
+      '',
+      `${red(
+        `Memory allocation of ${size} bytes failed at`
+      )} ${stringifyLocation(location)}`
+    );
+  }
+  return stringifyLines(lines);
+}
+
+function stringifyLocation(location: Location): string {
+  return typeof location === 'string'
+    ? green(location)
+    : stringifyGDBLine(location);
+}
+
+function stringifyGDBLine(gdbLine: GDBLine): string {
+  const { address, line } = gdbLine;
+  if (!isParsedGDBLine(gdbLine)) {
+    // Something weird in the GDB output format, report what we can
+    return `${green(address)}: ${line}`;
+  }
+  const filename = path.basename(gdbLine.file);
+  const dirname = path.dirname(gdbLine.file);
+  const file = `${dirname}${path.sep}${bold(filename)}`;
+  return `${green(address)}: ${blue(gdbLine.method, true)} at ${file}:${bold(
+    line
+  )}`;
+}
+
+function stringifyLines(lines: string[]): string {
+  return toTerminalEOL(lines.join(terminalEOL));
+}
+
+function toTerminalEOL(data: string): string {
+  return data.split(/\r?\n|\r/g).join(terminalEOL);
```

Comment:
Test this: `line1\rline2\rline3` should work.

When copy-pasting from the _Serial Monitor_ view, the content will be separated with `\r` "line endings". Check if this is a Theia bug or _Serial Monitor_ issue in IDE2.

Reply:
https://github.com/dankeboy36/esp-exception-decoder/pull/1/commits/993c409af9dc120ef3fd91d550970773ec1c3b97 and https://github.com/dankeboy36/esp-exception-decoder/pull/1/commits/bb1598cd86d73456717f5dda653fe4aac3034f5d